src/page-racial-disparities/PageRacialDisparities.js

```diff
@@ -0,0 +1,543 @@
+import { sum } from "d3-array";
+import React, { useState } from "react";
+import styled from "styled-components";
+import DetailPage from "../detail-page";
+import {
+  RACE_LABELS,
+  RACES,
+  SUPERVISION_TYPES,
+  TOTAL_KEY,
+  VIOLATION_COUNT_KEYS,
+  VIOLATION_TYPES,
+} from "../constants";
+import { Dropdown } from "../controls";
+import useChartData from "../hooks/useChartData";
+import Loading from "../loading";
+import { formatAsPct, sentenceCase } from "../utils";
+
+const ETHNONYMS = {
+  [RACES.nativeAmerican]: "people who are Native American",
+  [RACES.black]: "people who are Black",
+  [RACES.white]: "people who are white",
+  [RACES.hispanic]: "people who are Hispanic and/or Latino",
+  [RACES.other]: "people of other races",
+};
+
+const DynamicText = styled.span`
+  color: ${(props) => props.theme.colors.highlight};
+`;
+
+const BodySizeP = styled.p`
+  font: ${(props) => props.theme.fonts.body};
+
+  padding-right: calc(100% - ${(props) => props.theme.sectionTextWidthWide}px);
+`;
+
+const Footnote = styled.span`
+  font-size: 0.8em;
+  vertical-align: super;
+`;
+
+const FootnoteText = styled.p`
+  font-size: 0.8em;
+`;
+
+const matchRace = (race) => (record) => record.race_or_ethnicity === race;
+
+const getCorrectionsPopulation = (record) =>
+  record.total_incarcerated_population +
+  record.total_parole_population +
+  record.total_probation_population;
+
+const formatDecimal = (number) => number.toFixed(1);
+
+function getMetricsForGroup(data, category) {
+  const totals = data.find(matchRace(TOTAL_KEY));
+  const selected = data.find(matchRace(category));
+
+  const populationRate =
+    selected.total_state_population / totals.total_state_population;
+
+  const incarceratedPopulation = selected.total_incarcerated_population;
+  const incarceratedPopulationOverall = totals.total_incarcerated_population;
+  const parolePopulation = selected.total_parole_population;
+  const parolePopulationOverall = totals.total_parole_population;
+  const probationPopulation = selected.total_probation_population;
+  const probationPopulationOverall = totals.total_probation_population;
+
+  const supervisionPopulation = parolePopulation + probationPopulation;
+  const supervisionPopulationOverall =
+    parolePopulationOverall + probationPopulationOverall;
+
+  const correctionsPopulation = incarceratedPopulation + supervisionPopulation;
+  const correctionsPopulationOverall =
+    incarceratedPopulationOverall + supervisionPopulationOverall;
+
+  const correctionsPopulationRate =
+    correctionsPopulation / correctionsPopulationOverall;
+
+  const proportionIncarcerated = incarceratedPopulation / correctionsPopulation;
```

Comment:
We don't need to do any type-casting anywhere in here to ensure we get the division logic we want, right?

Reply:
no, we do need to typecast ... that was an oversight in my fake data